TypeCobol/Compiler/CodeElements/CodeElement.cs

```diff
@@ -29,6 +30,64 @@ protected CodeElement(CodeElementType type)
         /// </summary>
         public CodeElementType Type { get; }
 
+        /// <summary>
+        /// Describe how the CodeElement is debugged
+        /// </summary>
+        public DebugType DebugMode
+        {
+            get
+            {
+                if (_debugMode != DebugType.Unset) return _debugMode;
+                // detect CodeElement with a mix of Debug and "Normal" lines in debugging mode
+                int consumedTokensCount = ConsumedTokens.Count;
+                if (consumedTokensCount > 1)
+                {
+                    bool isDebug = false, isNoDebug = false;
+                    // CodeElement is on one line
+                    if (ConsumedTokens.First().Line == ConsumedTokens.Last().Line)
+                    {
+                        if (char.ToLower(ConsumedTokens[0].TokensLine.IndicatorChar) == 'd')
+                        {
+                            isDebug = true;
+                        }
+                        else
+                        {
+                            isNoDebug = true;
+                        }
+                    }
+                    else
+                    {
+                        // CodeElement is on multiple lines
+                        for (int i = 0; i < consumedTokensCount; i++)
+                        {
+                            bool isDebugType = char.ToLower(ConsumedTokens[i].TokensLine.IndicatorChar) == 'd';
+                            isDebug |= isDebugType;
```

Comment:
Please rename `isDebug` and `isNoDebug` to help understand the boolean tests

Reply:
Renamed to `atLeastOneDebug` and `atLeastOneWithoutDebug`.